File: Prácticas/P3/Java/civitas/src/civitas/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.Random;

/**
 *
 * @author Jose Padial
 */
public class Dado {
    private static Dado INSTANCE = new Dado();
    private static final int SalidaCarcel = 5;
    
    private Random random = new Random();
    private int ultimoResultado=0;
    private boolean debug = false;
    
    private Dado(){}
    
    public static Dado getInstance(){
        return INSTANCE;
    }
    
    int tirar(){
        if(debug == false){
            ultimoResultado = 1 + random.nextInt(6);
            return ultimoResultado;
        }
        return ultimoResultado=1;
    }
    
    boolean salgoDeLaCarcel(){
        return tirar() == SalidaCarcel;
    }
    
    int quienEmpieza(int n){
        return random.nextInt(n);
    }

    public void setDebug(boolean d) {
        Diario dia= Diario.getInstance();
        if(d){
            debug=true;
            dia.ocurreEvento("Debug activado");
        }else{
            debug=false;
            dia.ocurreEvento("Debug desactivado");
        }
    }

    int getUltimoResultado() {
        return ultimoResultado;
    }      
}

File: Prácticas/P2/Ruby/civitas/lib/dado.rb
# encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require "singleton"

module Civitas
  class Dado
    include Singleton
    SALIDA_CARCEL = 5
    
    attr_writer :debug
    attr_reader :ultimo_resultado
    def initialize
      @ultimo_resultado
      @debug = false
    end
    
    def tirar
      if @debug == false
        @ultimo_resultado = Random.rand(1..6)
      else
        @ultimo_resultado = 1
      end
      @ultimo_resultado
    end
    
    def salgo_de_la_carcel
      tirar() == SALIDA_CARCEL
    end
    
    def quien_empieza(n)
      Random.rand(1..n)
    end
  end
end
